sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -2190,9 +2192,9 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
                         progressHandler,
                         async,
                         cancellationToken).ConfigureAwait(false),
-                CommitPartitionedUpload = async (partitions, args, async, cancellationToken)
+                CommitPartitionedUpload = async (partitions, initialPosition, args, async, cancellationToken)
                     => await client.CommitBlockListInternal(
-                        partitions.Select(partition => StorageExtensions.GenerateBlockId(partition.Offset)),
+                        partitions.Select(partition => StorageExtensions.GenerateBlockId(partition.Offset - initialPosition)),
```

Comment:
I'm struggling to wrap my head around this line in particular.  Why does the blockId change based on the `initialPosition`?  This whole PR is kind of like mapping between coordinate systems.  We've got the data position relative to the local `Stream` and the data position relative to the blob in the cloud.  I would have thought partitions would be in blob units rather than `Stream` units but maybe that's where I'm getting hung up?